worked on tags, tags single file, date modified and some scss stuff

SeniorMars opened the pull request:
- **fixed tiny bug with clipboard**
- **removed unneeded css**
- **worked on tags, tags single file, date modified and some scss stuff**
- **added feature to hide toc with click/hover**
- Also, ran linter on some files and edited markdown.md

Apologies since there are supposed to be separate PRs but I working on this at the same time.


Solves: #53

SeniorMars commented:
I'll squash in a bit lol.

SeniorMars commented:
this is awful, sorry. Please when you are about to github's ""Squash and Merge"". BUt this is good work.

not-matthias commented on sass/parts/_header.scss:
```diff
@@ -1,77 +1,112 @@
+    /* Allows items to wrap as needed */
+}
+
+.meta>time {
+    color: var(--primary-color);
 }
```
I'm not sure if I like this: 
![image](https://github.com/not-matthias/apollo/assets/26800596/2a93361d-f921-47fd-b4a5-5a426da53f27)

Would you mind changing/removing this?

not-matthias commented on templates/macros/macros.html:
```diff
@@ -104,9 +103,25 @@ <h1 class="title">
                     {% endif %}
 
+                    {% if page.updated %}
+                        :: :odified on <time>{{ page.updated | date(format="%Y-%m-%d") }}</time>
```
Small typo here, should be 'M'odified :)

not-matthias commented:
Thanks for all the new features, I really like them. Just some minor changes and then we can merge it :)

SeniorMars commented:
Okay, so:

1. Modified the .meta tags to be something like this (with the pink removed!)
![CleanShot 2024-06-18 at 10 13 40@2x](https://github.com/not-matthias/apollo/assets/31679231/bb8a9430-4fdf-41c1-bdcc-80f08b50d1b3)

to enable something like this we need 

`repo_url = "https://github.com/SeniorMars/seniormars.com/blob/main/content/"` in config.toml and then

in your article:

```
[extra]
repo_view = true
read_time = true
```

this comes from `https://github.com/SeniorMars/redux/blob/0b6dcb48f74a24b4eaac9052db96d4ef6a984774/templates/macros/macros.html#L124` and `https://github.com/SeniorMars/redux/blob/0b6dcb48f74a24b4eaac9052db96d4ef6a984774/templates/macros/macros.html#L110`. I made sure to error handle.


 
 then I renamed `clipboard` to `fancy_code` as I realized that the code label also needed some js to move the label when the code is too long.
 
![CleanShot 2024-06-18 at 10 17 58@2x](https://github.com/not-matthias/apollo/assets/31679231/d53e72ee-5f63-489a-b3ea-22add6c1253b)

Furthermore, I added `h_anchor_link = true` which generates anchor tags for h1, h2, etc. I made sure to disable everything by default. 

Finally, I fixed some CSS and some unexpected behavior.

not-matthias commented:
Looks good! Thanks :)
